### PR TITLE
Disable core shadow presets by default, let themes opt-in

### PR DESCRIPTION
**Pull request by madhusudhand**
## What?

This PR 
* disables the default core presets, and themes needs to opt-in (feedback from: #58298)
* introduces 'unset` option to unset an existing shadow.

image

## Testing Instructions

* By default, shadows dropdown should not show core preset shadows. It should only show `unset` option.
* Enable default shadows in the active theme via `theme.json` and verify core presets in the dropdown.

```
{
    "settings": {
        "shadow": {
			"defaultPresets": false,
        }
   }
}
```

### CODE REVIEW

**Comment by madhusudhand** [comment]
@jasmussen @richtabor Can you suggest UI/copy when core presets are disabled and theme doesn't provide any presets for shadows?

image


**Duotone Reference:**

image

**Comment by jasmussen** [comment]
> Can you suggest UI/copy when core presets are disabled and theme doesn't provide any presets for shadows?

I'll defer to Rich if he has any input, but I'd simply omit the entire shadow control, and if need be, the entire panel, if there are no shadows defined.

**Comment by carolinan** [comment]
I agree with this change. Most importantly, 1) it doesn't break existing buttons where the user has saved a shadow. 2) The shadow can still be reset.

Kindly, inform me if this is planned to be included in 6.5. I will open new issues for the bundled themes, but it is a matter of when those issues need to be fixed.

**Comment by carolinan** [comment]
> > Can you suggest UI/copy when core presets are disabled and theme doesn't provide any presets for shadows?
> 
> I'll defer to Rich if he has any input, but I'd simply omit the entire shadow control, and if need be, the entire panel, if there are no shadows defined.

Removing the control or panel would prevent resetting, and the user would be stuck with their shadow if their theme is not updated to opt-in.

**Comment by vcanales** [comment]
> > Can you suggest UI/copy when core presets are disabled and theme doesn't provide any presets for shadows?
> 
> I'll defer to Rich if he has any input, but I'd simply omit the entire shadow control, and if need be, the entire panel, if there are no shadows defined.

cc. @richtabor, for visibility.

> Removing the control or panel would prevent resetting, and the user would be stuck with their shadow if their theme is not updated to opt-in.

This is a good point. I think we could maintain the reset button even if we hide the rest of the controls, but it might confuse the experience — ie. why is there a reset button but no shadow controls?

**Comment by jasmussen** [comment]
In absence of better suggestions, at least doing the same as Duotone does here may be the path forward for now. Then improve both, across, in a holistic way, separately.

**Comment by madhusudhand** [comment]
@jasmussen Introduced the unset option for shadows and is now ready for review.

image

**Comment by jasmussen** [comment]
I have to look at some other things today, so I'm happy to defer to others. But immediately the square shape stands out to me, and opens the question of whether the shadow presets should be circular and the same size as duotone, or colors as well. I think @richtabor may have some context there, so I'll defer to him.

**Comment by madhusudhand** [comment]
Also, should we introduce `clear` button too?

image

**Comment by jasmussen** [comment]
It seems like the same use case would exist for that, yes.

**Comment by richtabor** [comment]
>  But immediately the square shape stands out to me, and opens the question of whether the shadow presets should be circular and the same size as duotone, or colors as well. I think @richtabor may have some context there, so I'll defer to him.

I don't leveraging the existing duotone UI (circles, with the same unset control).

**Comment by richtabor** [comment]
> Removing the control or panel would prevent resetting, and the user would be stuck with their shadow if their theme is not updated to opt-in.

What's the scenario here? If someone has applied a core shadow at the global level? Why not **enable shadow all the time in global styles (like all other controls) and omit it at the block level if there are no shadow presets**? 

> This is a good point. I think we could maintain the reset button even if we hide the rest of the controls, but it might confuse the experience — ie. why is there a reset button but no shadow controls?

Yes. The big difference is that we do have core duotone presets... so the control is useful out of the box. 

If a theme does not add their own shadows (I wouldn't expect existing themes to do so much) then the component looks broken and effectively provides no value—just confusion. 

The current "always enable the component" flow makes much more sense when the shadow UI affords _creating_ shadows. Then it's added value to the experience.

**Comment by carolinan** [comment]
> Why not enable shadow all the time in global styles (like all other controls) and omit it at the block level if there are no shadow presets?

I did not understand what this means.
The option was enabled by default. When it is disabled by default, not all users will be able to enable it because they would have to use child themes or filters.

**Comment by richtabor** [comment]
> The option was enabled by default. When it is disabled by default, not all users will be able to enable it because they would have to use child themes or filters.

You can't add a shadow to a block, if there are no presets.

**Comment by carolinan** [comment]
> > The option was enabled by default. When it is disabled by default, not all users will be able to enable it because they would have to use child themes or filters.
> 
> You can't add a shadow to a block, if there are no presets.

No, but if you already have added them you get locked in.

**Review by vcanales — APPROVED** [review]

**Review by t-hamano — CHANGES_REQUESTED** [review]
I think we also need to update the schema in theme.json.

https://github.com/WordPress/gutenberg/blob/5939c41670df4f2caee40c24806d3be20a09dab1/schemas/json/theme.json#L80

**Comment by madhusudhand** [comment]
> I think we also need to update the schema in theme.json.

@t-hamano updated the scheme and docs. Thank you.

**Comment by t-hamano** [comment]
Thank you for the update! This PR seems to have been approved once, but my understanding is that there is still discussion about whether to show the panel, even if the `shadow` was not opted in, to avoid lock-in.

@richtabor @carolinan Is my understanding correct?

**Comment by carolinan** [comment]
I did not understand what Rich suggest as an alternative to keeping the reset.

Shadows were added in 6.2, so there will be an unknown amount of users -**on block themes only**  that have used the option and may find it unexpected that the option is there but the presets are now missing. They can still for example duplicate their button block if they want to re-use the style, but they won't know how to remove the style without editing the code or deleting the blocks.

**Comment by carolinan** [comment]
In classic themes, the scenario is different. They never had shadow support without theme.json, so there is no global style for the button block to reset, either.

-Users of classic themes will see a panel labeled "Border and shadow": [Even if border is not supported in the theme or the block.](https://github.com/WordPress/gutenberg/issues/58907)
-There are no presets, so they can not apply a shadow. They can not opt-in to the default shadow presets without creating a hybrid theme with theme.json.
-They can not add their own custom presets.
-They can not completely opt-out and remove the empty control.

**Comment by carolinan** [comment]
Question (not trying to sound like criticism, its an actual question): How come we opt out of the presets, not the feature?

**Comment by colorful-tones** [comment]
Hi folks, I believe merging this PR would likely close out this issue https://github.com/WordPress/gutenberg/issues/58907, which is currently on the [WordPress 6.5 Editor Tasks board](https://github.com/orgs/WordPress/projects/148/views/1), correct?

**Comment by madhusudhand** [comment]
> Question (not trying to sound like criticism, its an actual question): How come we opt out of the presets, not the feature?

Opting the feature is controlled by block settings such as `block.json` and currently the future is enabled for `button`, `image`, `column` and `columns` blocks using this setting. 

Where as, opting out default presets is done via `settings => shadow => defaultPresets: false` in theme.json and it removes the preset shadows from UI. 

Both of these settings are controllable, but they independent of theme type (classic or block). 
Introducing the conditional check based on the theme type here #58967

**Comment by madhusudhand** [comment]
> Hi folks, I believe merging this PR would likely close out this issue #58907, which is currently on the [WordPress 6.5 Editor Tasks board](https://github.com/orgs/WordPress/projects/148/views/1), correct?

@colorful-tones Sorry, This PR wouldn't fix the issue. I will add a separate fix.

**Comment by madhusudhand** [comment]
@richtabor please let us know if we are good to land this change.

**Comment by carolinan** [comment]
> > Question (not trying to sound like criticism, its an actual question): How come we opt out of the presets, not the feature?
> 
> Opting the feature is controlled by block settings such as `block.json` and currently the future is enabled for `button`, `image`, `column` and `columns` blocks using this setting.
> 
> Where as, opting out default presets is done via `settings => shadow => defaultPresets: false` in theme.json and it removes the preset shadows from UI.
> 
> Both of these settings are controllable, but they independent of theme type (classic or block). Introducing the conditional check based on the theme type here #58967

Yes but other block supports also need to be added in block.json, are disabled by default, and enabled either by appearanceTools or settings.category.feature.true I mean as in spacing.padding.true

**Comment by richtabor** [comment]
> Shadows were added in 6.2, so there will be an unknown amount of users -on block themes only that have used the option and may find it unexpected that the option is there but the presets are now missing. They can still for example duplicate their button block if they want to re-use the style, but they won't know how to remove the style without editing the code or deleting the blocks.

Shadows was only released in global styles, not in the button block inspector.

**Comment by richtabor** [comment]
If we merge this as-is, this is the default experience that effectively all users of block themes will see: 

CleanShot 2024-02-13 at 10 27 52

What I'm suggesting is that if there are no shadow presets available, we don't show the controls — it's just a "Border" `PanelBody`, with no tool to add a shadow for (as we don't support shadow _creation_ yet). 

And in Global Styles—where we have an existing "Effects" `PanelBody`, with a "Shadow" control—the control is rendered in like above (which is already the case in WordPress). This way, users can "unset" an existing globally-applied shadow (though I'd expect this to be such a small group of users). 

We should not be delivering a unfruitful experience (the above) for everyone, to support the very edge case of those users who discovered the shadow presets on the button block, in Global Styles, applied it holistically to all buttons, then want to remove the applied shadow on a specific button.

**Comment by carolinan** [comment]
> Shadows was only released in global styles, not in the button block inspector.

I did not say it was outside the global styles.

**Comment by carolinan** [comment]
So in simpler words:
The Styles > Blocks > Button should have the reset.
The block editor, for all supported blocks, should not show the control.
We would not be able to remove the shadow from **individual** button blocks, and this is a minor inconvenience compared to showing the reset control for all blocks.
Did I understand that correctly?

_When I write that I don't understand, it literally means that I don't understand, not that I think the person is wrong._

File: packages/block-editor/src/components/global-styles/shadow-panel-components.js
Comment (madhusudhand) [inline packages/block-editor/src/components/global-styles/shadow-panel-components.js]: I can see that settings in global styles always comes with `appearanceTools` set to false and in block style settings this key is not set. 
By adding a condition on this, adds unset option only in global styles as per [this](https://github.com/WordPress/gutenberg/pull/58766#issuecomment-1942015584) expectation. 

Need feedback on this condition.

Comment (madhusudhand) [reply packages/block-editor/src/components/global-styles/shadow-panel-components.js]: @t-hamano could you share feedback on this.

**Comment by t-hamano** [comment]
I re-read this PR and the discussion that took place in #58967.

Are the specifications of Shadow and what this PR aims for as follows? Please let me know if I'm wrong.

- In WP6.4, the shadow UI is only available in the global styles.
- To allow users to reset the shadow that they have already applied in a global style, this PR always provides an "unset" option for global styles, regardless of whether they have preset or not.
- In WP6.5, the shadow UI can be used at the block level as well.
- At the block level, if there is no preset, the shadow control itself is not displayed.
- If users have the latest Gutenberg plugin enabled in WP6.4 and were applying shadows from the default preset at the block level, this PR change will remove the default preset and the shadow UI itself will no longer be displayed. In this case, users cannot remove the shadow style unless they delete the block or use the code editor. However, this is considered as an edge case.

If all of this is correct, I think the focus is on **whether to take into account users who have already applied the shadow in the global styles**.

- If we support this case, this PR approach should be correct. However, there should no longer be a way to hide the shadow UI in the global styles.
- If we don't consider it, I think you need to publish this PR change as a dev note and encourage users to opt in to `defautPresets`.

cc @aaronrobertshaw @andrewserong

**Comment by andrewserong** [comment]
> If we don't consider it, I think you need to publish this PR change as a dev note and encourage users to opt in to defautPresets.

Good idea to flag the changes in a dev note. To make it easier for themes that wish to opt in to this to opt in, would it be worth adding `array( 'shadow', 'defaultPresets' )` to [the list of `APPEARANCE_TOOLS_OPT_INS`](https://github.com/WordPress/gutenberg/blob/ccfb9ff2d7ee0bc0dd52c7687eff37c2d5598704/lib/class-wp-theme-json-gutenberg.php#L649) so that the presets still default to `false`, but in themes that opt in to appearance tools, they don't need to update manually to still have access to the shadow controls? That might help mitigate any surprises about missing controls on themes that usually expect to get all the new controls added automatically.

**Comment by t-hamano** [comment]
> To make it easier for themes that wish to opt in to this to opt in, would it be worth adding `array( 'shadow', 'defaultPresets' )` to [the list of `APPEARANCE_TOOLS_OPT_INS`](https://github.com/WordPress/gutenberg/blob/ccfb9ff2d7ee0bc0dd52c7687eff37c2d5598704/lib/class-wp-theme-json-gutenberg.php#L649) so that the presets still default to `false`, but in themes that opt in to appearance tools, they don't need to update manually to still have access to the shadow controls?

I think this is a good idea! Even if this PR changes the default shadow preset to `false`, the global styles shadow UI should still look exactly the same if the user already has `appearanceTools` set to true, right?

The only concern is in cases where the user has not defined `appearanceTools` itself or has set `appearanceTools` to `false` and is utilizing shadows in the global styles. In this case, the shadow UI will not be visible in the global styles unless they set `appearanceTools` or `settings.shadow.defaultPresets` to `true`, but I think such users will probably be few and far between and can be covered in the dev notes.

I think this approach also eliminates the need to consider whether the shadow UI is being rendered within the global styles, as [this comment](https://github.com/WordPress/gutenberg/pull/58766#discussion_r1494446350) concerns.

**Comment by andrewserong** [comment]
> Even if this PR changes the default shadow preset to false, the global styles shadow UI should still look exactly the same if the user already has appearanceTools set to true, right?

Yes, I believe that's correct 👍. The default will still be `false`, but then [this bit of code](https://github.com/WordPress/gutenberg/blob/ccfb9ff2d7ee0bc0dd52c7687eff37c2d5598704/lib/class-wp-theme-json-gutenberg.php#L768-L774) will internally switch it to `true` if it's in the list of appearance tools and the theme has opted-in for `appearanceTools`. It should work the same as the other properties that default to `false` in the core `theme.json` like `spacing.margin` and `spacing.padding`.

**Comment by aaronrobertshaw** [comment]
Thanks for the ping 👍 

I don't have much to add. The current plan around leveraging `appearanceTools` to smooth out the transition aligns with what I proposed in the other PR.

> I think such users will probably be few and far between and can be covered in the dev notes.

Agreed, I don't think this edge case is something that the UI has to optimise for.

My only other thoughts are around if there is a way we could capture some of the learnings gained here from the introduction of the shadow support to make future additions easier. Any ideas?

**Comment by carolinan** [comment]
If the theme does not add appearance tools, but adds custom shadows, will the shadow control still be available?
I mean, the appearance tools addition will still only be for the default shadow presets, not the feature?

**Comment by carolinan** [comment]
This is not a request but a point of confusion:
>I mean, the appearance tools addition will still only be for the default shadow presets, not the feature?

**Comment by t-hamano** [comment]
> If the theme does not add appearance tools, but adds custom shadows, will the shadow control still be available?

Yes, that's my understanding.

This is because whether the shadow control feature is enabled or not depends on whether default shadow presets or custom shadow presets are present.

File: packages/block-editor/src/components/global-styles/border-panel.js
Comment (t-hamano) [inline packages/block-editor/src/components/global-styles/border-panel.js]:
```suggestion
	const shadows = getShadowPresets( settings );
	return !! settings?.shadow?.length;
```

It should be possible to simplify it like this. Also, considering the consistency throughout this file, it may be better to move this function under the `useHasBorderWidthControl` function.

File: packages/block-editor/src/components/global-styles/shadow-panel-components.js
Comment (t-hamano) [inline packages/block-editor/src/components/global-styles/shadow-panel-components.js]: Since this component will only be called once here, we might want to hardcode the component here.

File: packages/block-editor/src/components/global-styles/shadow-panel-components.js
Comment (t-hamano) [inline packages/block-editor/src/components/global-styles/shadow-panel-components.js]: It would be better to use `classnames` library

File: packages/block-editor/src/components/global-styles/shadow-panel-components.js
Comment (t-hamano) [inline packages/block-editor/src/components/global-styles/shadow-panel-components.js]: If we didn't care about global styles, we could write this function much more simply.

I haven't tested it, but we should be able to write it like this: Additionally, memorize the return value and use an `EMPTY_ARRAY` variable.

```javascript
export function getShadowPresets( settings ) {
	if ( ! settings?.shadow ) {
		return EMPTY_ARRAY;
	}

	const defaultPresetsEnabled = settings?.shadow?.defaultPresets;
	const { default: defaultShadows, theme: themeShadows } = settings?.shadow;
	const unsetShadow = {
		name: __( 'Unset' ),
		slug: 'unset',
		shadow: 'none',
	};

	return useMemo(
		() => [
			unsetShadow,
			...( themeShadows || EMPTY_ARRAY ),
			...( ( defaultPresetsEnabled && defaultShadows ) || EMPTY_ARRAY ),
		],
		[ defaultPresetsEnabled, defaultShadows, themeShadows ]
	);
}
```

File: packages/block-editor/src/components/global-styles/shadow-panel-components.js
Comment (t-hamano) [inline packages/block-editor/src/components/global-styles/shadow-panel-components.js]: This class name should not be necessary

Comment (madhusudhand) [reply packages/block-editor/src/components/global-styles/shadow-panel-components.js]: Removed it. ✅

File: packages/block-editor/src/components/global-styles/shadow-panel-components.js
Comment (madhusudhand) [reply packages/block-editor/src/components/global-styles/shadow-panel-components.js]: updated the function with similar code structure and wrapped entire function call inside `useMemo`

File: packages/block-editor/src/components/global-styles/shadow-panel-components.js
Comment (madhusudhand) [reply packages/block-editor/src/components/global-styles/shadow-panel-components.js]: Updated the with `classnames` call

File: packages/block-editor/src/components/global-styles/shadow-panel-components.js
Comment (madhusudhand) [reply packages/block-editor/src/components/global-styles/shadow-panel-components.js]: moved the component code inline.

File: packages/block-editor/src/components/global-styles/border-panel.js
Comment (madhusudhand) [reply packages/block-editor/src/components/global-styles/border-panel.js]: Could not merge because they are two different variables `settings.shadow` and `shadows`. Moved the function below `useHasBorderWidthControl` as suggested.

Comment (t-hamano) [reply packages/block-editor/src/components/global-styles/border-panel.js]: Oh I missed the "s" 😅

**Comment by t-hamano** [comment]
@madhusudhand Thanks for the update!

Based on the discussion so far, I think there is no need to consider whether it is a global style or not.

In other words, this part should no longer be necessary.

https://github.com/WordPress/gutenberg/pull/58766/files#diff-2879a8cc0f5dedde1efb617e374f4926a3eb8d25cf68eff3c0577f66963325a4R160-R161

Instead, I think we can just add `array( 'shadow', 'defaultPresets' )` to `APPEARANCE_TOOLS_OPT_INS`, as mentioned in [this comment](https://github.com/WordPress/gutenberg/pull/58766#issuecomment-1958663732). What do you think?

**Comment by andrewserong** [comment]
> In other words, this part should no longer be necessary.

I suppose the main thing to make sure is that the `unsetShadow` isn't showing up when it shouldn't be. It looks like the logic of `useShadowPresets` will include `unsetShadow` even if `defaultPresetsEnabled` is `false`.

I wonder if, alongside adding `defaultPresets` to the list of appearance tools, if it'd work for the `unsetShadow` to only be injected if there's at least one preset? So perhaps something like:

```
		const shadowPresets = [
			...( ( defaultPresetsEnabled && defaultShadows ) || EMPTY_ARRAY ),
			...( themeShadows || EMPTY_ARRAY ),
		];
		if ( shadowPresets.length ) {
			shadowPresets.unshift( unsetShadow );
		}
		return shadowPresets;
```

Then we might not need the `isGlobalStyles = settings?.hasOwnProperty( 'appearanceTools' );` line?

Apologies if I'm missing some nuance there!

**Comment by t-hamano** [comment]
> if it'd work for the `unsetShadow` to only be injected if there's at least one preset?

Indeed, I think this is the ideal behavior. The code I suggested in [this comment](https://github.com/WordPress/gutenberg/pull/58766#discussion_r1498828893) was not sufficient 😅

**Comment by madhusudhand** [comment]
@andrewserong @t-hamano @aaronrobertshaw Thank you for the suggestions. 

Following is the current behavior.

1. `appearanceTools: false`, shadows panel is hidden
2. `appearanceTools: false` but theme sets presets via theme.json `shadow: { presets: [{}] }` , then shadow panel would appear to show theme shadows

image

3. `appearanceTools: true`, then core presets enabled.

image

defaultPresets can be explicitly be set to false to hide presets in this case.

In any case, unset shadow options would appear only when there is at least one shadow.

@carolinan Please share your feedback.

File: packages/block-editor/src/components/global-styles/shadow-panel-components.js
Comment (t-hamano) [inline packages/block-editor/src/components/global-styles/shadow-panel-components.js]:
```suggestion
			settings?.shadow?.presets ?? {};
```

This should prevent a critical error that occurs when shadows are enabled _only_ on certain blocks.

See the screencast below.

https://github.com/WordPress/gutenberg/assets/54422211/e7aade5c-04f6-49e7-94f1-515c9257df7d

**Comment by t-hamano** [comment]
I've tried various scenarios and everything seems to be working as expected, except for what I mentioned in [this comment](https://github.com/WordPress/gutenberg/pull/58766#discussion_r1500458845).

Also, I think this PR is going to be backported to WP6.5, so I would like to give it the necessary labels.

File: packages/block-editor/src/components/global-styles/shadow-panel-components.js
Comment (madhusudhand) [reply packages/block-editor/src/components/global-styles/shadow-panel-components.js]: Great catch. Thank you!

**Comment by carolinan** [comment]
_Not a code review_

I have my opinions, but I am able to set them aside. ;) 
I tested some edge cases like a block theme without theme.json, classic theme with appearance-tools theme support, and classic theme with theme.json, and the control displays or hides correctly.

**Review by t-hamano — APPROVED** [review]
LGTM! My understanding is that this PR should have settled on the most ideal approach 👍 There is one conflict in the style sheet, but I think it is okay to accept both competing styles.

Thanks again for your efforts!

**Comment by swissspidy** [comment]
> I've tried various scenarios and everything seems to be working as expected, except for what I mentioned in [this comment](https://github.com/WordPress/gutenberg/pull/58766#discussion_r1500458845).
> 
> Also, I think this PR is going to be backported to WP6.5, so I would like to give it the necessary labels.

This PR is labelled as an enhancement and we're past the enhancement stage for 6.5. Can you elaborate why it is important to have this in 6.5 still?

**Comment by madhusudhand** [comment]
The core fix was to change the core preset shadows from `opt-out` to `opt-in` by default, which is a bug from last release.
However, a new option `unset` to reset a shadow which came as part of the discussion as the fix changes the behavior of the dropdown. And so it was marked as enhancement.

As the root problem is a bug fix, I guess we may want to change it to Bug fix instead?

**Comment by getdave** [comment]
To me this feels like a Bug Fix. It seems originally introduced prior to 6.5 but I'd be inclined to include it if it produces a better experience for users in 6.5. What do you think @swissspidy?

Also pinging @annezazu and @fabiankaegy as Editor Triage leads.

We don't have long here as packages will need to be synced shortly.

**Comment by swissspidy** [comment]
I trust your judgement 👍

**Comment by fabiankaegy** [comment]
@getdave i agree on this. Let's include it

**Comment by getdave** [comment]
Ok @madhusudhand let's bring this in.

Perhaps @swissspidy can approve your [Core Backport PR](https://github.com/WordPress/wordpress-develop/pull/6178)?

Also please could you test the bug fix works as expected during the [Beta 3 release party](https://make.wordpress.org/core/2024/02/01/6-5-release-parties-schedule-and-hosts-2/) and subsequent release?

**Comment by getdave** [comment]
I just cherry-picked this PR to the cherry-pick-wp-6-5-beta-3 branch to get it included in the next release: a446441007

**Comment by annezazu** [comment]
I'm also in agreement that this feels bug territory for the record. Thanks for handling this so promptly!

File: lib/theme.json
Comment (ajlende) [inline lib/theme.json:194]: @madhusudhand This is a breaking change for themes that have shadow presets with the same slug as the core presets as it changes the CSS output for the theme. It should have a theme.json version bump and migration to maintain backwards compatibility.

I've been discussing this in https://github.com/WordPress/gutenberg/pull/58409#issuecomment-1997603910 since I want to change the default for the font sizes presets as well.

Comment (madhusudhand) [reply lib/theme.json:194]: This change updates `defaultPresets` from true to false, and also it is controlled by appearanceTools setting.
Meaning that defaultPresets are off by default and when `appearanceTools` is set to true, they are enabled, which makes existing themes that are using default presets to work as before.

Also, I created #59893 to discuss on this idea of theme being able to add(override) a preset with same slug as core preset.

Comment (ajlende) [reply lib/theme.json:194]: > Meaning that `defaultPresets` are off by default and when appearanceTools is set to true, they are enabled, which makes existing themes that are using default presets to work as before.

`appearanceTools` is also `false` by default, so it's still a problem for themes that haven't configured either.

Before this change, with a theme that has custom `settings.shadow.presets` defined with matching slugs to those in the defailt lib/theme.json and `settings.shadow.defaultPresets` and `settings.appearanceTools` both undefined, you'll end up with different CSS that gets output to the page.

In the first case the default styles will be output and in the second case, the theme styles will be output.

Here's an example of what I'm talking about.

```jsonc
// Relevant parts of theme.json
{
  "settings": {
    // `appearanceTools` is not set
    "shadow": {
      // `defaultPresets` is not set
      "presets": [
        {
          "name": "Natural",
          "slug": "natural",
          "shadow": "2px 2px 3px #000"
        }
      ]
    }
  }
}
```

```css
/* Generated CSS before */
--wp--preset--shadow--natural: 6px 6px 9px rgba(0, 0, 0, 0.2);
```

```css
/* Generated CSS after */
--wp--preset--shadow--natural: 2px 2px 3px #000;
```

It's because to the `settings.shadow.defaultPresets` being used as the value for `prevent_override` in `PRESETS_METADATA` in `WP_Theme_JSON`. The setting defines the behavior of matching slugs in the default and theme origins for presets in theme.json.

https://github.com/WordPress/gutenberg/blob/cdfd75183952ce7ae1ec78e673ac719e002d2de8/lib/class-wp-theme-json-gutenberg.php#L181

https://github.com/WordPress/gutenberg/blob/cdfd75183952ce7ae1ec78e673ac719e002d2de8/lib/class-wp-theme-json-gutenberg.php#L2868-L2876

I'm suggesting that we revert this for 6.5 (at least this line since it is the one causing the problems). Then we can use the theme.json version bump to opt-in to breaking changes like this.

I'm adding the theme.json version bump in #58409. So you'd just need to update the migration from that PR to make sure that existing themes are unaffected.

Hopefully that all makes sense. 😅

Comment (madhusudhand) [reply lib/theme.json:194]: Thank for details @ajlende. 
I see that it may break a shadow when the following conditions are met 
* `appearanceTools: false`
* theme defined custom shadows 
* and they are defined with same slug as defaultPresets

Also, prior to 6.5 shadows are enabled for `button` block and in global styles only.
It seems to be very unlikely edge case and shouldn't have big impact on themes.

@annezazu @swissspidy @t-hamano what do you think?

Comment (getdave) [reply lib/theme.json:194]: Also cc @fabiankaegy editor triage co-lead.

Comment (getdave) [reply lib/theme.json:194]: I decided it was best to create a dedicated issue where this can be discussed.

https://github.com/WordPress/gutenberg/issues/59989

This is also useful as it can be placed in `Triage` on the [WP 6.5 Editor board](https://github.com/orgs/WordPress/projects/148) for review by Editor triage leads.

**Comment by juanmaguitar** [comment]
@Leonardus-Nugraha was there a DevNote published for this PR?
I see the `has-dev-note` label but I haven't been able to find the related DevNote.
I also don't find this PR tracked at [6.5 Dev Note Tracking Issue](https://github.com/WordPress/gutenberg/issues/59155) or at [WP 6.5 Documentation](https://github.com/orgs/WordPress/projects/154) project dashboard
cc: @madhusudhand

**Comment by Leonardus-Nugraha** [comment]
@juanmaguitar this was initially a part of the [Miscellaneous Editor Changes dev note](https://make.wordpress.org/core/2024/03/09/miscellaneous-editor-changes-in-wordpress-6-5), but it was removed as @madhusudhand informed that the [change was reverted](https://github.com/WordPress/wordpress-develop/pull/6309).

Convo here: https://wordpress.slack.com/archives/C065MBW03EH/p1711618356661949?thread_ts=1711107767.424069&cid=C065MBW03EH

**Comment by juanmaguitar** [comment]
Thanks for the info @Leonardus-Nugraha! 
As part of GB 17.9, I tracked this PR at [6.6 Dev Note Tracking Issue #62096](https://github.com/WordPress/gutenberg/issues/62096). I'll remove it from [this tracking issue](https://github.com/WordPress/gutenberg/issues/62096) then, as I understand no DevNote is needed anymore.

**Pull request merged**
